vmm/src/device_manager/mmio.rs

```diff
@@ -312,10 +313,22 @@ mod tests {
             format!("{}", e),
             "unable to add device to kernel command line: string contains an equals sign"
         );
-        let e = Error::CloneIoeventFd(sys_util::Error::new(0));
-        assert_eq!(format!("{}", e), "failed to clone ioeventfd: Error(0)");
-        let e = Error::CloneIrqFd(sys_util::Error::new(0));
-        assert_eq!(format!("{}", e), "failed to clone irqfd: Error(0)");
+        let e = Error::CloneIoeventFd(io::Error::from_raw_os_error(0));
+        assert_eq!(
+            format!("{}", e),
+            format!(
+                "failed to clone ioeventfd: {:?}",
```

Comment:
Use the actual error text and get rid of the `format!`:

```suggestion
                "failed to clone ioeventfd: No error information (os error 0)",
```

Reply:
Looks like this won't work exactly the same way between gnu and musl (maybe even others).

I'll revert this to using format by `{}` instead for now.

```rust
test result: ok. 23 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running /tmp/firecracker_test_session/firecracker_binaries/test/x86_64-unknown-linux-gnu/debug/deps/vmm-b33a6f9ca44b473a

running 45 tests
// snip
test device_manager::mmio::tests::test_error_messages ... FAILED
// snip

     Running /tmp/firecracker_test_session/firecracker_binaries/test/x86_64-unknown-linux-musl/debug/deps/vmm-b33a6f9ca44b473a

running 47 tests
// snip
test device_manager::mmio::tests::test_error_messages ... ok
// snip
```